build.gradle.kts

```diff
@@ -295,8 +296,91 @@ subprojects {
                 sign(publishing.publications["shadow"])
             }
         }
-    }
+    } else {
+        apply(plugin = "signing")
+        apply(plugin = "maven-publish")
+        apply(plugin = "net.linguica.maven-settings")
+
+        val sourcesJar by tasks.registering(Jar::class) {
+            archiveClassifier.set("sources")
+            from(sourceSets.main.get().allSource)
+        }
+
+        val javadocJar by tasks.registering(Jar::class) {
+            archiveClassifier.set("javadoc")
+            from(tasks["javadoc"])
+        }
```

Comment:
Given the above block (300->312) is almost a duplicate of the lines around 162 I would think we could improve the if blocks?

Reply:
I am still seeing what looks like duplicate definitions of ```
        apply(plugin = "signing")
        apply(plugin = "maven-publish")
        apply(plugin = "net.linguica.maven-settings")
``` and sourcesJar, javadocJar, etc ? 


So while functionally it looks correct, I think it could be optimised?